### PR TITLE
WIP -- DO NOT MERGE Edit profile

### LINKED ISSUE #38: Add edit user profile feature

### PR DESCRIPTION
**Pull request by mngatewood**
- [x] Wrote Tests
- [x] Implemented
- [x] Reviewed

# Necessary checkmarks:
- [x] All Tests are Passing
- [x] The code will run locally

## Type of change
- [x] New feature
- [ ] Bug Fix

# Implements/Fixes:  Adds edit profile feature
## Description of Changes
_adds tests
_adds flash message container to application.html
_adds update method to users controller

closes #38 

# Check the correct boxes
- [x] This broke nothing
- [ ] This broke some stuff
- [ ] This broke everything

# Testing Changes
- [ ] No Tests have been changed
- [x] Some Tests have been changed
- [ ] All of the Tests have been changed(Please describe what in the world happened)

# Checklist:

- [x] My code has no unused/commented out code
- [x] I have reviewed my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have fully tested my code

# Please include a link to a gif of how you feel about this branch:

### CODE REVIEW

File: spec/features/user_updates_profile_spec.rb [hunk 1]
```diff
@@ -20,17 +20,57 @@
...
       visit profile_edit_path
 
+      # no changes made to any fields
+
+      click_on 'Update User'
+
+      save_and_open_page
+
       within("main.update-user") do
         expect(current_path).to eq(profile_edit_path)
```
[the remark this answers is not in the document]
Comment (mnhollandplum) [reply spec/features/user_updates_profile_spec.rb]: @mnhollandplum

[the remark this answers is not in the document]
Comment (mnhollandplum) [reply spec/features/user_updates_profile_spec.rb]: @iandouglas @dionew1 can you take a look at the comment and test?

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -22,11 +22,19 @@ def edit
...
+    if  params[:user][:password].blank? || 
+        params[:user][:password_confirmation].blank?
+    # Otherwise, remove both from params
+      params[:user].delete(:password)
+      params[:user].delete(:password_confirmation)
+    end
+    if current_user.update_attributes(user_params)
+      flash[:success] = "Your profile has been updated."
+      redirect_to profile_path
     else
```
Comment (mngatewood) [inline app/controllers/users_controller.rb]: @iandouglas @dionew1 

I can get in here with a pry in development by loading the page and clicking Update User without changing any of the fields.  The test should do the same (visit profile_edit_path, click_on 'Update User'), but it redirects to profile_path and flashes a success message.

Comment (iandouglas) [reply app/controllers/users_controller.rb]: Are you mocking the login in your test? If so, the current_user object won't return any chances you make to the user.

File: spec/features/user_updates_profile_spec.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (iandouglas) [reply spec/features/user_updates_profile_spec.rb]: If the edit page is loaded and they don't make any changes, how are you detecting that they've made no changes to the object? (I wouldn't bother with checking if they HAVE changed anything, just save whatever gets submitted unless there's an error in the data, then report on that)

**Pull request merged**